### PR TITLE
docs: update instructions for Techdocs setup with Docker

### PR DESCRIPTION
**Pull request by plektra**
## Hey, I just made a Pull Request!

Bringing the instructions up to date for setting up Techdocs in Docker container. The latest Debian-based images require Python packages to be installed in an isolated environment. Also bumping mkdocs-techdocs-core version to the latest.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ x All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/techdocs/getting-started.md [hunk 1]
```diff
@@ -263,22 +263,33 @@ Setting `generator.runIn` to `local` means you will have to make sure your
...
 ```Dockerfile
-RUN apt-get update && apt-get install -y python3 python3-pip
-RUN pip3 install mkdocs-techdocs-core==1.1.7
+RUN apt-get update && apt-get install -y python3 python3-pip python3-venv
+
+ENV VIRTUAL_ENV=/opt/venv
+RUN python3 -m venv $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+RUN pip3 install mkdocs-techdocs-core==1.2.3
```
Comment (awanlin) [inline docs/features/techdocs/getting-started.md:279]: Would it make sense not to pin the version and then mention how to pin and the [changelog](https://github.com/backstage/mkdocs-techdocs-core#changelog) that has the versions below this instead?

File: docs/features/techdocs/getting-started.md [hunk 2]
```diff
@@ -263,22 +263,33 @@ Setting `generator.runIn` to `local` means you will have to make sure your
...
 ```
 
 Please be aware that the version requirement could change, you need to check our
 [`Dockerfile`](https://github.com/backstage/techdocs-container/blob/main/Dockerfile)
 and make sure to match with it.
 
-Note: We recommend Python version 3.7 or higher.
+On a Debian-based Docker container, Python packages must be either installed using
+the OS package manager or within a virtual environment (see the
+[related PEP](https://peps.python.org/pep-0668/)). Alternative is to use e.g.
```
Comment (awanlin) [inline docs/features/techdocs/getting-started.md:288]: You'll need to add PEP and pipx to the `accepted.txt` - https://github.com/backstage/backstage/blob/master/.github/vale/Vocab/Backstage/accept.txt - to make vale happy

**Review by awanlin — APPROVED** [review]
Thanks again @plektra, looks great! 🚀

**Review by benjdlambert — APPROVED** [review]
Thanks :tada: e2es failing is unrelated to this PR.

**Pull request merged**